Umee-Bots/Lending-Pool-Reentrancy/src/agent.ts

```diff
@@ -0,0 +1,42 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+
+import CONFIG from "./agent.config";
+
+import utils, { AgentConfig } from "./utils";
+
+export const provideHandleTransaction = (config: AgentConfig): HandleTransaction => {
+  const sigHashes = utils.getSigHashes(config.reentrancyBlacklist);
+
+  const handleTransaction: HandleTransaction = async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const { traces } = txEvent;
+    for (let i = 0; i < traces.length; i++) {
+      const trace = traces[i];
+      const depth = trace.traceAddress.length;
+      if (trace.action.to === config.lendingPoolAddress) {
+        let alerted = false;
+        for (let j = i + 1; j < traces.length; j++) {
+          if (traces[j].traceAddress.length <= depth) {
+            i = j - 1;
+            break; // subtree ended, non reentrant call
+          }
+
+          if (traces[j].action.to === config.lendingPoolAddress && !alerted) {
+            const selector = (traces[j].action.input || "").slice(0, 10); // "0x" and first 4 bytes
+            if (sigHashes.includes(selector)) {
+              const initialCallSelector = (txEvent.transaction.data || "").slice(0, 10);
+              findings.push(utils.createFinding(initialCallSelector, selector));
+              alerted = true;
+            }
+          }
```

Comment:
```suggestion
        for (i = i + 1; i < traces.length; i++) {
          if (traces[i].traceAddress.length <= depth) {
            i--;
            break; // subtree ended, non reentrant call
          }

          if (!alerted && traces[i].action.to === config.lendingPoolAddress) {
            const selector = (traces[i].action.input || "").slice(0, 10); // "0x" and first 4 bytes
            if (sigHashes.includes(selector)) {
              const initialCallSelector = (trace.action.input || "").slice(0, 10);
              findings.push(utils.createFinding(initialCallSelector, selector));
              alerted = true;
            }
          }
```
This fixes the unexpected test result mentioned in a previous comment. Also, about `initialCallSelector`, since the call it's referring to is the one from `trace`, the selector should be from `trace` input, not the transaction data.